typed/measures-with-dimensions/chemistry/compound-struct.rkt

```diff
@@ -16,15 +16,17 @@ require racket/match
   ([alist : (Listof (Pairof (Un Element Compound) Natural))])
   #:transparent
   #:property prop:custom-write
-  (lambda ([comp : Compound] [out : Output-Port] [mode : (Un 0 1 #t #f)])
+  (lambda ([comp : Compound] [out : Output-Port] [mode : (Un Boolean 0 1)])
     (match mode
-      [_ #:when (andmap zero? (map cdr (compound-alist comp)))
+      [_ #:when (andmap zero? (map (lambda ([x : (Pairof (Un Element Compound) Natural)])
+                                     (cdr x))
+                                   (compound-alist comp)))
```

Comment:
I would prefer something like `(inst cdr (Un Element Compound) Natural)` instead of this lambda.

Reply:
Cool. I didn't know that. Thank you for the tip